### PR TITLE
 add scheduler for optimization

### PR DESCRIPTION
**Pull request by oneseer**
add scheduler.py and test_scheduler.py
@fzliu

### CODE REVIEW

**Comment by oneseer** [comment]
import torch raise runtime error:
torch overrides.py '_has_torch_function' already has a docstring
check this issue
